HOWTO-make-a-release.md

```diff
@@ -0,0 +1,377 @@
+send the announcement emails later.  Keep a note of those commands for
+future reference.
+
+Verify that the tarballs are available via FTP:
+
+        ftp://ftp.openssl.org/source/
+
+And that they are ready for the website:
+
+        ls /var/www/openssl/source
+
+*For OpenSSL 3.0 and on*, push your local changes to the public source repo
+as instructed by `dev/release.sh`.  You may want to sanity check the pushes
+by inserting the `-n` (dry-run) option.
+
+*For OpenSSL before 3.0*, simply push your local changes to the public
+source repo, and please do remember to push the release tags as well, which
+is done separately with the `--tags` option.  You may want to sanity check
+the pushes by inserting the `-n` (dry-run) option.
+
+## Updating the website
+
+Push the website changes you made earlier to the OpenSSL website repo.  When
+you do this, the website will get updated and a script to flush the Akamai CDN
+cache will be run.  You can look at things on www-origin.openssl.org; the
+CDN-hosted www.openssl.org should only be a few minutes delayed.
+
+# Post-publishing tasks
+
+## Check the website
+
+Verify that the release notes, which are built from the CHANGES file in the
+release, have been updated.  This is done automatically by the commit-hook,
+but if you see a problem, try the following steps on `dev.openssl.org`:
+
+    cd /var/www/openssl
+    sudo -u openssl -H make relupd
+    sudo -u openssl ./bin/purge-one-hour
+
+Wait for a while for the Akamai flush to work (normally within a few minutes).
+Have a look at the website and news announcement at:
+
+-   <https://www.openssl.org/>
+-   <https://www.openssl.org/news/>
+
+Check the download page has updated properly:
+
+-   <https://www.openssl.org/source/>
+
+Check the notes look sensible at:
+
+-   <https://www.openssl.org/news/newslog.html>
+
+Also check the notes here:
+
+-   <https://www.openssl.org/news/openssl-1.0.2-notes.html>
+-   <https://www.openssl.org/news/openssl-1.1.0-notes.html>
+-   <https://www.openssl.org/news/openssl-1.1.1-notes.html>
+
+## Send the announcement mail
+
+Send out the announcements.  Generic release announcement messages will be
+created automatically by the build script and the commands you need to use
+to send them were displayed when you executed do-release.pl above.
+These should normally be sent from the openssl account.  These are sent to
+openssl-users, openssl-project, and openssl-announce.
+
+If do-release.pl was used with `--move` be sure to move the announcement
+text files away from the staging directory after they have been sent.  This
+is done as follows (with VERSION replaced with the version of OpenSSL to
+announce):
+
+        sudo -u openssl \
+            mv ~openssl/dist/new/openssl-VERSION.txt.asc ~openssl/dist/old
+
+Send out the Security Advisory if there is one.  Copy the file to the
+openssl user home directory, and then do the following:
+
+        sudo -u openssl gpg -u 8B3D79F5 --clearsign secadv_FILENAME
```

Comment:
I didn't do this step this time because it wasn't relevant, but I noticed that this tells you to use the OpenSSL user's GPG key. But I always sign security advisories with the same key that I sign releases (i.e. with my personal gpg key).

Reply:
Yeah...  so should I simply remove `-u 8B3D79F5`?

Reply:
Well, for me at least I always do the signing on my own machine and not on the dev box, so I think the text needs a bit of adjustment.

Reply:
I simply removed the `-u` option and argument

Reply:
I'm not sure that's accurate. Now the text says to copy the file to the openssl users home directory and sign it via `sudo -u openssl` (without a `-H` I might note), so that doesn't work if you want to sign using your own key. You should instead sign it *before* transferring it to the openssl user (and sudo is not necessary assuming you have privs to access your own key).

Reply:
Good point!

Reply:
I rewrote this quite a bit, please have a look